### PR TITLE
fix sscanf in cups for systems with different local number format

### PR DESCRIPTION
**Pull request by rslsp1**
The HWMargins are not correctly imported from the PPD on systems with non-English number formats.

### CODE REVIEW

File: src/common/cups_print.c [hunk 1]
```diff
@@ -61,6 +61,29 @@ typedef struct dt_prtctl_t
...
+      n++;
+    }
+  return n;
+}
+
+void convert_str_to_loc_numbers (char *data)
+{
+  struct lconv *currentLocalConv;
+  currentLocalConv = localeconv ();
+  char loc_decimal_point = currentLocalConv->decimal_point[0];
```
Comment (TurboGit) [inline src/common/cups_print.c:82]: const

File: src/common/cups_print.c [hunk 2]
```diff
@@ -61,6 +61,29 @@ typedef struct dt_prtctl_t
...
+    {
+      *ix++ = rep;
+      n++;
+    }
+  return n;
+}
+
+void convert_str_to_loc_numbers (char *data)
+{
+  struct lconv *currentLocalConv;
```
Comment (TurboGit) [inline src/common/cups_print.c:80]: const struct lconv *currentLocalConv = localeconv();

File: src/common/cups_print.c [hunk 3]
```diff
@@ -61,6 +61,29 @@ typedef struct dt_prtctl_t
...
+  currentLocalConv = localeconv ();
+  char loc_decimal_point = currentLocalConv->decimal_point[0];
+  char en_decimal_point = '.';
+  replacechar(data,en_decimal_point,loc_decimal_point);
```
[the remark this answers is not in the document]
Comment (rslsp1) [reply src/common/cups_print.c]: Thanks, done

File: src/common/cups_print.c [hunk 4]
```diff
@@ -61,6 +61,29 @@ typedef struct dt_prtctl_t
...
+    }
+  return n;
+}
+
+void convert_str_to_loc_numbers (char *data)
+{
+  struct lconv *currentLocalConv;
+  currentLocalConv = localeconv ();
+  char loc_decimal_point = currentLocalConv->decimal_point[0];
+  char en_decimal_point = '.';
```
[the remark this answers is not in the document]
Comment (rslsp1) [reply src/common/cups_print.c]: done

File: src/common/cups_print.c [hunk 5]
```diff
@@ -61,6 +61,29 @@ typedef struct dt_prtctl_t
...
+      *ix++ = rep;
+      n++;
+    }
+  return n;
+}
+
+void convert_str_to_loc_numbers (char *data)
+{
+  struct lconv *currentLocalConv;
+  currentLocalConv = localeconv ();
```
[the remark this answers is not in the document]
Comment (rslsp1) [reply src/common/cups_print.c]: done

File: src/common/cups_print.c (hunk 1 above)
Comment (rslsp1) [reply src/common/cups_print.c]: done

File: src/common/cups_print.c (hunk 2 above)
Comment (rslsp1) [reply src/common/cups_print.c]: done

Comment (rslsp1) [reply src/common/cups_print.c]: I have renamed convert_str_to_loc_numbers to dt_convert_str_to_loc_numbers
hope this is Ok.

**Review by TurboGit — CHANGES_REQUESTED** [review]
Ok, one last point to avoid duplicating this generic code. I'll place this routine into common/utility.{h,c} and rename it `dt_util_str_to_loc_numbers() `

**Comment by rslsp1** [comment]
Ok move to utility and rename done

**Review by TurboGit — APPROVED** [review]
Thanks !

**Comment by johnny-bit** [comment]
After merging this seems to cause CI issues.

it's this code:

```
/home/runner/work/darktable/darktable/src/src/common/utility.c:705:42: error: implicit declaration of function ‘localeconv’ [-Werror=implicit-function-declaration]
  705 |   const struct lconv *currentLocalConv = localeconv ();
      |                                          ^~~~~~~~~~
compilation terminated due to -Wfatal-errors.
```

**Pull request merged**